From 386d61d3440b8e97224b980c68985b5d3939d585 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Tue, 7 Feb 2012 15:19:16 +0100
Subject: [PATCH] dracut: only source "install", if available

If "install" is not there or executable, do not source it.
---
 dracut | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dracut b/dracut
index eb989db..4693261 100755
--- a/dracut
+++ b/dracut
@@ -243,7 +243,7 @@ for moddir in "$dracutbasedir/modules.d"/[0-9][0-9]*; do
 	if [[ $kernel_only = yes ]]; then
 	    [[ -x $moddir/installkernel ]] && . "$moddir/installkernel"
 	else
-	    . "$moddir/install"
+	    [[ -x "$moddir/install" ]] && . "$moddir/install"
 	    if [[ $no_kernel != yes && -x $moddir/installkernel ]]; then
 		. "$moddir/installkernel"
 	    fi
